cmd/auth-handler.go

```diff
@@ -106,6 +108,8 @@ func getRequestAuthType(r *http.Request) authType {
 		return authTypePostPolicy
 	} else if _, ok := r.Header["Authorization"]; !ok {
 		return authTypeAnonymous
+	} else if _, ok := r.URL.Query()["Action"]; ok {
+		return authTypeSTS
```

Comment:
`authTypeSTS` doesn't seemed to be used anywhere?

Reply:
It's meant for differentiation other than anonymous.

Reply:
Did not get it.

Did you mean to do 
```
	} else if _, ok := r.Header["X-Amz-Security-Token"]; ok {
		return authTypeSTS
```

Reply:
No it is an introduction to differentiate the authTypes might be used specific to STS API some require proper auth when action is specified.

Reply:
Since STS is anonymous, redirect handler uses authTypeAnonymous to redirect browser requests, we need to differentiate this. It is also useful in the long run as well when we introduce other STS APIs